src/promptflow/tests/sdk_cli_test/e2etests/test_tool.py

```diff
@@ -0,0 +1,296 @@
+        tool_meta = self.get_tool_meta(tool_path)
+        expect_tool_meta = {
+            "test_tool.tool_with_custom_strong_type_connection.my_tool": {
+                "name": "Tool With Custom Strong Type Connection",
+                "type": "python",
+                "inputs": {
+                    "connection": {"type": ["CustomConnection"], "custom_type": ["MyCustomConnection"]},
+                    "input_text": {"type": ["string"]},
+                },
+                "description": "This is my tool with custom strong type connection.",
+                "module": "test_tool.tool_with_custom_strong_type_connection",
+                "function": "my_tool",
+            }
+        }
+        assert tool_meta == expect_tool_meta
+
+    def test_tool_with_input_settings(self):
+        tool_path = TOOL_ROOT / "tool_with_dynamic_list_input.py"
+        tool_meta = self.get_tool_meta(tool_path)
+        expect_tool_meta = {
+            "test_tool.tool_with_dynamic_list_input.my_tool": {
+                "name": "My Tool with Dynamic List Input",
+                "type": "python",
+                "inputs": {
+                    "input_text": {
+                        "type": ["list"],
+                        "is_multi_select": True,
+                        "allow_manual_entry": True,
+                        "dynamic_list": {
+                            "func_path": "test_tool.tool_with_dynamic_list_input.my_list_func",
+                            "func_kwargs": [
+                                {
+                                    "name": "prefix",
+                                    "type": ["string"],
+                                    "reference": "${inputs.input_prefix}",
+                                    "optional": True,
+                                    "default": "",
+                                },
+                                {
+                                    "name": "size",
+                                    "type": ["int"],
+                                    "optional": True,
+                                    "default": 10
+                                },
+                            ],
+                        },
+                    },
+                    "input_prefix": {"type": ["string"]},
+                },
+                "description": "This is my tool with dynamic list input",
+                "module": "test_tool.tool_with_dynamic_list_input",
+                "function": "my_tool",
+            }
+        }
+        assert tool_meta == expect_tool_meta
+
+        tool_path = TOOL_ROOT / "tool_with_enabled_by_value.py"
+        tool_meta = self.get_tool_meta(tool_path)
+        expect_tool_meta = {
+            'test_tool.tool_with_enabled_by_value.my_tool': {
+                'name': 'My Tool with Enabled By Value',
+                'type': 'python',
+                'inputs': {
+                    'user_type': {'type': ['string']},
```

Comment:
The `user_type` doesn't have enum value here ? it should include enum: - student; - teacher

Reply:
Fix it, update test case